hypershift-operator/controllers/hostedcluster/apiproviders/platform_reconciler.go

```diff
@@ -0,0 +1,33 @@
+package apiproviders
+
+import (
+	"context"
+	"fmt"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+)
+
+const (
+	HostedClusterAnnotation = "hypershift.openshift.io/cluster"
+)
+
+// PlatformReconciler is an interface for a platform specific reconciler.
+type PlatformReconciler interface {
+	ReconclieCred(ctx context.Context, hcluster *hyperv1.HostedCluster, cpns string) error
+	ReconcileSecret(context.Context, *hyperv1.HostedCluster, string) error
+	GetInfraCR(context.Context, *hyperv1.HostedCluster, *hyperv1.HostedControlPlane, string) (client.Object, error)
```

Comment:
This is called get GetInfraCR but it's actually doing more, should probably be `ReconcileInfrastructureClusterCR`